### PR TITLE
feat: Adds more slots to the modal + makes it more robust

### PR DESCRIPTION
**Pull request by ChristianKienle**

### CODE REVIEW

File: ui/src/components/ClickAwayContainer.ts [hunk 1]
```diff
@@ -52,31 +52,36 @@ export default Vue.extend({
...
-        }
-        if (isActive && !wasActive) {
-          this.$el.addEventListener("click", this.click, false);
-          documentElement.addEventListener("click", this.click, false);
-        }
-        if (!isActive && wasActive) {
-          this.$el.removeEventListener("click", this.click, false);
-          documentElement.removeEventListener("click", this.click, false);
-        }
+      async handler(isActive: boolean, wasActive: boolean): Promise<void> {
```
Comment (xujing-shen) [inline ui/src/components/ClickAwayContainer.ts]: can we remove async here?

File: ui/src/components/ClickAwayContainer.ts [hunk 2]
```diff
@@ -52,31 +52,36 @@ export default Vue.extend({
...
+        setTimeout(() => {
```
Comment (xujing-shen) [inline ui/src/components/ClickAwayContainer.ts]: is nextTick() better than setTimeout() here?

File: ui/src/components/ClickAwayContainer.ts (hunk 1 above)
Comment (ChristianKienle) [reply ui/src/components/ClickAwayContainer.ts]: Good catch. You are correct. My intention was to promisify setTimeout but I never did it so async can go. Thanks!

Comment (ChristianKienle) [reply ui/src/components/ClickAwayContainer.ts]: @xujing-shen Just double checked it. `$nextTick` seems not to be enough.

File: ui/src/components/ClickAwayContainer.ts (hunk 2 above)
Comment (ChristianKienle) [reply ui/src/components/ClickAwayContainer.ts]: @xujing-shen Just double checked it. `$nextTick` seems not to be enough.

**Pull request merged**